activerecord/test/cases/relation/where_chain_test.rb

```diff
@@ -75,7 +80,11 @@ def test_missing_with_multiple_association
     def test_missing_merged_with_scope_on_association
       # This query does not make much logical sense, but it is testing
       # that the generated SQL query is valid.
-      assert_equal Author.find(1).posts.count, Post.where.missing(:author).merge(Author.where(id: 1)).count
+      assert_equal Author.find(1).posts.count, Post.where.associated(:author).merge(Author.where(id: 1)).count
```

Comment:
I think this one was changed by accident?

Reply:
Yes this was, apologies again and thanks for your patience.

Reply:
Just squashed everything into one branch with a helpful description of bug and fixes!  Thanks for your help.